src/NuGet.Clients/NuGet.Console/PowerConsoleToolWindow.cs

```diff
@@ -522,7 +522,7 @@ private async Task OnDispatcherStartCompletedAsync()
             await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
What is the difference between this JoinableTaskFactory and the default one?

Reply:
I'm sure Nikolche will correct me if I'm wrong, but I believe that when the solution contains a CPS project, then we use the CPS JTF, rather than the NuGetPackage's JTF.  Why that happens, I have no idea. That was implemented before I joined the team.

Reply:
Best description in: https://github.com/NuGet/NuGet.Client/blob/f6e5a71a4d1ac6fd4f461cfe0097c7a6f4496c2e/src/NuGet.Clients/NuGet.VisualStudio.Common/NuGetUIThreadHelper.cs#L40-L62

I know this is a long standing things to avoid certain deadlocks, something that members of the team worked on with Lifeng and Andrew.